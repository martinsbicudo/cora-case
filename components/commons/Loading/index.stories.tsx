import { StoryObj, Meta } from '@storybook/react'

import Loading from '.'

export default {
  title: 'Commons / Loading',
  component: Loading,
} as Meta

export const Default: StoryObj = {}
