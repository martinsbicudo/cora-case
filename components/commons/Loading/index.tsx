import * as Styled from './styles'

const Loading = () => {
  return (
    <Styled.Loading>
      <div />
      <div />
      <div />
      <div />
    </Styled.Loading>
  )
}

export default Loading
