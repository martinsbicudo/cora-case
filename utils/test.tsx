import type { ReactElement, ReactNode } from 'react'

import { ThemeProvider } from 'styled-components'

import theme from '@/styles/theme'
import { render, RenderOptions } from '@testing-library/react'

const Providers = ({ children }: { children: ReactNode }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: Providers, ...options })

export * from '@testing-library/react'
export { customRender as render }
