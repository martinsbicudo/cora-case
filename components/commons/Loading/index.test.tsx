import { render } from '@/test/helpers'

import Loading from '.'

describe('Components / Loading', () => {
  const renderComponent = () => render(<Loading />)

  it('should render component', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })
})
