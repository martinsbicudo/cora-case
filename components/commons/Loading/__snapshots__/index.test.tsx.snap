// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components / Loading should render component 1`] = `
<div>
  <div
    class="sc-bcXHqe cIKesE"
  >
    <div />
    <div />
    <div />
    <div />
  </div>
</div>
`;
